andy_elf/8259pic/8259pic.c

```diff
@@ -35,8 +35,9 @@ void legacy_pic_remap(int offset1, int offset2)
 	outb(PIC2_DATA, ICW4_8086);
 	io_wait();
 
-	outb(PIC1_DATA, a1);   // restore saved masks.
-	outb(PIC2_DATA, a2);
+	outb(PIC1_DATA, 0);   // enable all.
+	io_wait();
+	outb(PIC2_DATA, 0);
   kputs("done.\r\n");
```

Comment:
minor - should probably `io_wait` here as well just in case the next instruction also sends to the PIC